File: mywebroot/CS471/NOTES/PYTHON/iterators.rb
#!/usr/bin/ruby
# sample iterator code in Ruby

module IteratorEx
	def iteratorB()  #() are note required.
		yield 3
		yield 4
	end

	def iteratorC()
		yield 11
		yield 22
		yield 33
	end
    
	def iteratorA(i)
		yield i*10
	end
	def nestedLoop()
		iteratorC() {|i| iteratorA(i) do |x|
						iteratorB() do |y|
							print x , ', ', y ,"\n"
						end
					end
				   }
	end
	def fibLazy(upto)
		fb1 = 1 #fn-2
		fb2 = 1 #fn-1
		oldfb1 =0
		while fb1 <= upto 
			(fb2,fb1,oldfb1) = fb1+fb2, fb2, fb1
			yield oldfb1
		end
	end
	
end

include IteratorEx

iteratorB{|i| print i; print "\n"}

iteratorB(){|i| print i; print "\n"}

iteratorC(){|i| print i; print "\n"}

nestedLoop(){|i| put "Nested Loop"}

fibLazy(100) {|fib|  puts fib}

# Scripts illustrating numbers are objects not primitives

sum = 0
1.upto(10){|n| sum = sum +n}
puts sum

puts "LIST"
lst = [ ]
0.upto(10){|n| lst[n]=n}
print lst
print "\n"

File: mywebroot/CS471/HW/1Experments/iterators.py
def iteratorB():
    yield 3
    yield 4


# f = iteratorC() --- f.next()
def iteratorC():
    yield 1
    yield 2
    yield 3
def iteratorA(iter):
    for i in iter:
        yield i*10
    

#call loop( iteratorA() )
def loop(iter):
    for i in iter:
        print repr(i)

        
def nestedLoop():
    for x in iteratorA(iteratorC()):
        for y in iteratorB():
            print repr(x)
            print repr(y)

def fibs():
    fb1 = 1 #fn-2
    fb2 = 1 #fn-1
    while True:
        (fb1,fb2,oldfb2) = (fb1+fb2, fb1, fb2)
        yield oldfb2
